lib/codecs/h264.js

```diff
@@ -16,6 +16,13 @@ NalByteStream = function() {
     buffer;
   NalByteStream.prototype.init.call(this);
 
+  /*
+   * Scans a byte stream and triggers a data event with the NAL units found.
+   * @param {Object} data Event received from H264Stream
```

Comment:
This may not always come from the H264Stream, it's just the only source right now. It may be a bit "dangerous" to comment about a specifically named component from elsewhere, even though they're coupled pretty tightly right now, because the comment may become out-of-date without someone realizing it's there.

It may also be better to not consider the event either, as that's an upstream concern around the stream concept.

Reply:
I think while NalByteStream is enclosed by H264Stream, it's highly unlikely that we would get other kinds of data. In fact, within this method we assume that we're dealing with a specific format(and box structure). 

To clarify further why I chose to mention H264Stream explicitly: we currently have zero indications that NalByteStream is packaged within H264Stream and they are coupled together. Without removing the Stream concept entirely I don't think we can stay away from referring to specific Streams as sources as it is not clear to an unfamiliar reader.

Personally, when I was new to the project, I had no clue what was going on here

>It may also be better to not consider the event either, as that's an upstream concern around the stream concept.

what do you mean by not consider?